api/accounts/index.ts

```diff
@@ -3,57 +3,59 @@ import * as express from 'express'
 import {getConfigValue} from '../configuration'
 import {SERVICES_FEE_AND_PAY_API_PATH} from '../configuration/references'
 import {FeeAccount} from '../interfaces/feeAccountPayload'
+import {valueOrNull} from '../lib/util'
 import {getAccountUrl} from './accountUtil'
 
 async function handleAddressRoute(req, res) {
-    let errReport: any
-    if (!req.query.accountNames) {
-        errReport = {
-            apiError: 'Account is missing',
-            apiStatusCode: 400,
-            message: 'Fee And Pay route error',
-        }
-        res.status(errReport.apiStatusCode).send(errReport)
+  let errReport: any
+  if (!req.query.accountNames) {
+    errReport = {
+      apiError: 'Account is missing',
+      apiStatusCode: 400,
+      message: 'Fee And Pay route error',
     }
-    const accountNames = req.query.accountNames.split(',')
-    const accounts = new Array<FeeAccount>()
-    const accountPromises = new Array<AxiosPromise<any>>()
+    res.status(errReport.apiStatusCode).send(errReport)
+  }
+  const accountNames = req.query.accountNames.split(',')
+  const accounts = new Array<FeeAccount>()
+  const accountPromises = new Array<AxiosPromise<any>>()
 
-    accountNames.forEach((accountNumber: string) => {
-        const feeAndPayHostUrl = getConfigValue(SERVICES_FEE_AND_PAY_API_PATH)
-        const url = getAccountUrl(feeAndPayHostUrl, accountNumber)
-        accountPromises.push(getAccountPromise(url, req))
-    })
-    let responseStatusCode
-    try {
-            await Promise.all(accountPromises).then(allAccounts => {
-                allAccounts.forEach(account => {
-                    if (account.status === 404) {
-                        responseStatusCode = 404
-                    }
-                    accounts.push(account.data)
-                })
-            })
-        } catch (error) {
-            errReport = {
-                apiError: error && error.data && error.data.message ? error.data.message : error,
-                apiStatusCode: error && error.status ? error.status : '',
-                message: `Fee And Pay route error `,
-            }
-            res.status(error.status).send(errReport)
-            return
+  accountNames.forEach((accountNumber: string) => {
+    const feeAndPayHostUrl = getConfigValue(SERVICES_FEE_AND_PAY_API_PATH)
+    const url = getAccountUrl(feeAndPayHostUrl, accountNumber)
+    accountPromises.push(getAccountPromise(url, req))
+  })
+  let responseStatusCode
+  try {
+    await Promise.all(accountPromises).then(allAccounts => {
+      allAccounts.forEach(account => {
+        if (account.status === 404) {
+          responseStatusCode = 404
         }
-    if (responseStatusCode) {
-        res.status(responseStatusCode).send(accounts)
-    } else {
-        res.send(accounts)
+        accounts.push(account.data)
+      })
+    })
+  } catch (error) {
+    const status = valueOrNull(error, 'status')
```

Comment:
@paddybasi  can you encapsulate this error handling code to a function?

Reply:
This PR is just to protect against a runtime TypeError. We can (and should!) do a further clean up task later, as @Mo-Lala84 mentioned previously. I would rather replace all of these with custom errors that we can throw, then catch in generic error handler